checker.cpp

```diff
@@ -1,46 +1,63 @@
 #include <iostream>
 #include <assert.h>
 #include <unistd.h>
+
 using namespace std;
 
-int vitalsOk(float temperature, float pulseRate, float spo2) {
-  if(temperature > 102 || temperature < 95) {
-    cout << "Temperature critical!\n";
-    for (int i = 0; i < 6; i++)
-    {
-      cout << "\r* " << flush;
-      sleep(1);
-      cout << "\r *" << flush;
-      sleep(1);
-    }
-
-    return 0;
-  } else if(pulseRate < 60 || pulseRate > 100) {
-    cout << "Pulse Rate is out of range!\n";
-    for (int i = 0; i < 6; i++)
-    {
-      cout << "\r* " << flush;
-      sleep(1);
-      cout << "\r *" << flush;
-      sleep(1);
-    }
-    return 0;
-  } else if(spo2 < 90) {
-    cout << "Oxygen Saturation out of range!\n";
-    for (int i = 0; i < 6; i++)
-    {
-      cout << "\r* " << flush;
-      sleep(1);
-      cout << "\r *" << flush;
-      sleep(1);
-    }
-    return 0;
-  }
-  return 1;
+void flush()
+{
+	for (int i = 0; i < 6; i++)
+	{
+		cout << "\r* " << flush;
+		sleep(1);
+		cout << "\r *" << flush;
+		sleep(1);
+	}
 }
 
-int main() {
+bool TemperatureCheck(float temperature)
+{
+	if (temperature > 102 || temperature < 95) 
+	{
+		cout << "Temperature critical!\n";
+		flush();
+		return false;
+	}
+	return true;
+}
+
+bool PulseCheck(float pulseRate)
+{
+	if (pulseRate < 60 || pulseRate > 100)
```

Comment:
Consider isolating the range-comparison in a separate function, to make it easier to implement warnings.